design/MultipleQueues.md

```diff
@@ -0,0 +1,83 @@
+# Multiple Queues
+
+Multiple queues is important for:
+
+- Explicit submission onto less capable queues, such as pure copy queues.
+- Explicit distribution of submissions on machines with multiple queues of the same family.
+
+## QueueFamily
+
+There are multiple families of queues.
```

Comment:
What is the benefit of exposing families? Why not make each queue independent, having its own capabilities?

Reply:
Because multiple queues from the same family do not need to transfer ownership between each other

Reply:
They also allow for less idling when complex execution dependencies are present
https://mynameismjp.wordpress.com/2018/06/17/breaking-down-barriers-part-3-multiple-command-processors/

Reply:
I don't think we've discussed the technique for ownership transfer. Depending on how that works, the implementation could be the one determining which barriers are issued, so it would know internally if the two queues happen to be in the same family and omit barriers accordingly.

Reply:
Yes, but this transfer will be costly so the user has to have exposed queue family information, so that they can avoid using resources across queues from different families.